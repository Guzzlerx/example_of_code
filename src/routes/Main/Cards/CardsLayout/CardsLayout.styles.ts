import styled from 'styled-components';

export const StyledContainer = styled.section<{ hasBreadCrumbs?: boolean }>`
  margin-top: ${props => (props.hasBreadCrumbs ? '32px' : '64px')};
  flex: 1;
  padding-left: 100px;
`;
