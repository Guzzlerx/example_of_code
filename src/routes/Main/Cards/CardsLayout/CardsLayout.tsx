import { Outlet } from 'react-router-dom';
import { StyledContainer } from './CardsLayout.styles';

export const CardsLayout = () => (
  <StyledContainer hasBreadCrumbs>
    <Outlet />
  </StyledContainer>
);
