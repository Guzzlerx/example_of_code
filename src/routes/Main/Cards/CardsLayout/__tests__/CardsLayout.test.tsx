import { RenderResult, render, screen } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import i18n from 'utils/languages/i18nForTests';
import { CardsLayout } from '../CardsLayout';

describe('CardsLayout', () => {
  let renderResult: RenderResult;

  beforeEach(() => {
    renderResult = render(
      <MemoryRouter initialEntries={['/cards/child']}>
        <I18nextProvider i18n={i18n}>
          <Routes>
            <Route path="cards" element={<CardsLayout />}>
              <Route path="child" element={<div>CHILD</div>} />
            </Route>
          </Routes>
        </I18nextProvider>
      </MemoryRouter>,
    );
  });

  it('should render correctly', () => {
    expect(renderResult.container).toBeInTheDocument();
  });

  it('snapshot', () => {
    expect(renderResult.container).toMatchSnapshot();
  });

  it('should render Outlet correctly', () => {
    const outlet = screen.getByText(/child/i);

    expect(outlet).toBeInTheDocument();
  });
});
