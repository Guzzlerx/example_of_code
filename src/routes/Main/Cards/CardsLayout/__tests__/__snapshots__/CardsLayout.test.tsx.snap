// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsLayout snapshot 1`] = `
<div>
  <section
    class="sc-bcXHqe gNJUCg"
  >
    <div>
      CHILD
    </div>
  </section>
</div>
`;
